src/lib/nvme_pt_ioctl.c

```diff
@@ -727,6 +738,51 @@ static int validate_sessn(struct opal_device *dev)
 	return 0;
 }
 
+static int parse_tper_host_prop(struct sed_device *device)
+{
+	struct opal_device *dev = device->priv;
+	int payload_len = dev->payload.len;
+	struct sed_tper_properties *tper = &device->discv.sed_tper_props;
+
+	/* TPer properties are returned as key-value pairs */
+	for (int i = 0, j = 0; i < payload_len; i++) {
+		if (resp_token_match(dev->payload.tokens[i], OPAL_STARTNAME)) {
+			int jmp = get_payload_string(dev, i + 1);
+			int key_len = dev->payload.tokens[i + 1]->len - jmp;
+
+			assert(j < 23);
+			memcpy(tper->property[j].key_name,
+				dev->payload.tokens[i + 1]->pos + jmp,
+				key_len);
+			tper->property[j].value =
+					dev->payload.tokens[i + 2]->vals.uint;
+			j++;
+		}
+	}
+
+	return 0;
```

Comment:
This function always return 0. why not to have it return void?

Reply:
yes!, can be changed to void